### ISSUE TITLE
Allow other GraphiQL interfaces

### ISSUE THREAD

**Issue by patrick91** [strawberry-graphql/strawberry#2943]
I think we should allow users to choose between GraphiQL and other interfaces (like the Apollo Explorer).

This might not be too difficult to implement now that we have a base view, but maybe we need some tweaks for Django, or we should at least consider to (keep) support(ing) overriding the playground using templates (I think this works now, right @bellini666?)

The only thing I'm not sure about this, is if we set this option on the schema, or on the views. The views currently also have the ability to enable/disable GraphiQL, so maybe it should live there.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/strawberry-graphql) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/strawberry-graphql/strawberry/issues/2943">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/2943/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/2943/pledge.svg">
</picture>
</a>

**Comment by bellini666** [comment]
> I think this works now, right @bellini666?

Yes, the django view [will try to load it from a  template](https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/django/views.py#L147), and then fallback to strawberry's one if no one is defined. Users can define their own by creating a `templates/graphql/graphiql.html` file in their apps

**Comment by patrick91** [comment]
@bellini666 thanks for checking 😊 do you think we should change the path if they chose a different interface?

For example:

```python
urlpatterns = [
    path("graphql/", GraphQLView.as_view(schema=schema, graphql_interface="apollo-explorer")),
]
```

this would use `templates/graphql/apollo-explorer.html`?

Also I think we need a nicer name rather than `graphql_interface` :D

**Comment by bellini666** [comment]
@patrick91 I think that is fine. I assume that not a lot of people do override default templates in django, as that is an "advanced operation". Those that do will probably know which one they should override, specially considering the intersection of people that will both "use another interface" and also "override it" =P

**Comment by devvspaces** [comment]
I think apollo interface is great

**Comment by DoctorJohn** [comment]
This has been resolved via #3209. We now support multiple GraphQL IDEs out of the box and allow rendering of custom IDEs.
